### ISSUE TITLE
axis names not properly exported in Map.write()

### ISSUE THREAD

**Issue by registerrier** [gammapy/gammapy#1429]
When using a map in regular (gadf) format (see http://gamma-astro-data-formats.readthedocs.io/en/latest/skymaps/index.html), the axis names are not properly exported in the BAND HDU. The columns are then AXIS0, AXIS0_MIN and AXIS0_MAX although the Primary header correctly states the name in the keyword AXLCOLS. 
 This causes a failure in `Map.read()` because  AXLCOLS is not consistent with the column names.

 The issue is likely caused by expected behavior when working with different format conventions (e.g. fgsst) which expect ENERGY, E_MIN, E_MAX column names rather than the ENERGY, ENERGY_MIN, ENERGY_MAX behavior expected when one follows the gadf convention.

An example:
```python
import numpy as np
from gammapy.maps import WcsNDMap,MapAxis, Map
axis0=MapAxis.from_edges(np.linspace(1,10),unit='',name='e_true')
mymap=WcsNDMap.create(skydir=(0,0),width=10,binsz=1,axes=[axis0])
mymap.write("test.fits")
Map.read("test.fits")
```
breaks with `ValueError: 'E_TRUE_MIN' is not in list`
The header of the BANDS extension  is the following:
```XTENSION= 'BINTABLE'           / binary table extension                         
BITPIX  =                    8 / array data type                                
NAXIS   =                    2 / number of array dimensions                     
NAXIS1  =                   14 / length of dimension 1                          
NAXIS2  =                   49 / length of dimension 2                          
PCOUNT  =                    0 / number of group parameters                     
GCOUNT  =                    1 / number of groups                               
TFIELDS =                    4 / number of table fields                         
AXCOLS1 = 'E_TRUE_MIN,E_TRUE_MAX'                                               
EXTNAME = 'PRIMARY_BANDS'      / extension name                                 
TTYPE1  = 'CHANNEL '                                                            
TFORM1  = 'I       '                                                            
TTYPE2  = 'AXIS0   '                                                            
TFORM2  = 'E       '                                                            
TTYPE3  = 'AXIS0_MIN'                                                           
TFORM3  = 'E       '                                                            
TTYPE4  = 'AXIS0_MAX'                                                           
TFORM4  = 'E       '
```

**Comment by registerrier** [comment]
The issue comes from `make_axes_cols` in `gammapy.maps.geom` which is called by `MapGeom.make_bands_hdu()`

I have tried to introduce a specific behavior by passing the convention to `make_axes_cols`. This solves the issue above but breaks the tests...

**Comment by cdeil** [comment]
Indeed, this seems to be a bug in `make_axes_cols`:
https://github.com/gammapy/gammapy/blob/c209481f42edf675dc5d675cb2fd4ffd1ec9873d/gammapy/maps/geom.py#L43

It happens when `map.write` is called for the case where an axis has been given an explicit name, that is different from the names `energy` and `time` that are special-cased. Like `name="e_true"` in the example by @registerrier above.

Here's a complete example with resulting FITS headers and stack trace:
https://gist.github.com/cdeil/08c54d4b2b349dd3c0a411ed488ba788

My understanding is that the problem is that currently this is written:
```
AXCOLS1 = 'E_TRUE_MIN,E_TRUE_MAX'                                               
TTYPE2  = 'AXIS0   '                                                            
TTYPE3  = 'AXIS0_MIN'                                                           
TTYPE4  = 'AXIS0_MAX'                                                           
```
but actually this should be written:
```
AXCOLS1 = 'E_TRUE_MIN,E_TRUE_MAX'                                               
TTYPE2  = 'E_TRUE   '                                                            
TTYPE3  = 'E_TRUE_MIN'                                                           
TTYPE4  = 'E_TRUE_MAX'                                                           
```

@woodmd - Can you confirm?

I'll try to have a closer look and attempt a fix.

**Comment by cdeil** [comment]
For me the small change in #1433 fixes the issue and makes all tests pass.
I.e. I think it was a but that's localised in `make_axes_cols` for the case of having a custom axis name other than `'energy'`, `'time'` or the empty string `''`.

**Comment by registerrier** [comment]
Issue solved by PR #1433 .
Closing issue.
